app/models/episode.rb

```diff
@@ -254,6 +254,10 @@ def apple_mark_for_reupload!
     apple_needs_delivery!
   end
 
+  def apple_episode
+    podcast.apple_config&.build_publisher&.show&.episodes&.find { |e| e.feeder_episode.id == id }
```

Comment:
So just tracing through this...

1. Get the `Apple::Config` for this podcast
2. Gets a new `Apple::Api` from that config
3. Gets a new `Apple::Publisher` with that api
4. Gets a new `Apple::Show` with that api
5. Gets many new `Apple::Episode` for most `Episode`s in the podcast
6. Filters those down to episodes in the apple subscription `Feed`
7. Finds this 1 episode within that

Kind of convoluted?  But doesn't seem like it makes any API requests along the way, so probably ok functionally.

Reply:
yes, I agree, it's convoluted, but that's the point of this helper, to jump through the hoops from the episode -> apple_episode in this one place.

I could break up those steps into additional lines or helper methods, but this is fairly concise?

Reply:
I do like how you kept it all together/concise.  Maybe just a comment, that this doesn't make API requests?  That's mainly why i traced through all that - wasn't obvious at face value if this was making remote calls.

Reply:
looking at this again, one part we may not need are steps 5 - 7.
once you have the `show`, you can construct an apple episode:
```
def apple_episode
    show = podcast.apple_config&.build_publisher&.show
    Apple::Episode.new(api: show.api, show: show, feeder_episode: self) if show
end
```
@svevang do you see an issue with skipping those steps and instantiating from the feeder episode and apple show?

Reply:
Up to this point there has been no ::Episode -> Apple::Episode logic, but ultimately we need to do a join episodes that are "apple synced" to this episode.

Per your comment @cavis 1,2,3 are concomitant/procedural and (in any case) can't really be split up, so maybe that could be a helper? Then a guard on the publisher `return nil unless apple_publisher.present?` or even `apple_show.present?`, since we'd be modeling this as a relation almost (not a publishing endeavor) .

Reply:
@kookster Instantiation sounds fine, but might not be what you want?

If this method is always returning an apple episode, then there is no way to indicate that this episode would never be considered for publishing to apple  like in the case of an episode that is not in the private feed.  We'd likely want to display no status in that case? (`Apple::Show#episodes` are the episodes in the private feed)

Reply:
There is a guard on the view logic before this is called, where it checks if there should be apple status:
https://github.com/PRX/feeder.prx.org/blob/main/app/views/episodes/_form_status.html.erb#L34
```
 <% if episode.persisted? && episode.publish_to_apple? %>
```

I could pull that in to this helper as well, so we don't try to find an apple episode if there should not be one.

Reply:
`Episode::publish_to_apple?` might need to be tweaked a bit too. https://github.com/PRX/feeder.prx.org/blob/cbd501db3cd912d742e3cacbd5984f75cf80e0e2/app/models/concerns/apple_delivery.rb#L20-L22 as it's just wrapping the apple_config check (not incorporating information about which feed the episode is in)

Looking at https://github.com/PRX/feeder.prx.org/blob/cbd501db3cd912d742e3cacbd5984f75cf80e0e2/app/models/apple/show.rb#L208-L217 Is it too onerous to keep leveraging this? So that we don't mirror the interface / have to handle the edge cases that this solves for (like an episode that's in a third unrelated feed would never be published to apple).

Reply:
(Related: we're dangerously close to having a `feed.episodes` association in #1019 - would simplify finding episodes that should be in this `Feeds:AppleSubscription`)

Reply:
Direct instantiation would work with a join the Feeds::AppleSubscription, though I think we'd need a new branch in the switch

```
if apple_episode.nil?
  'not found'
elsif apple_episode.apple_new?
  'new'
```
And maybe a branch for apple archived?

Joining to the Apple::Episode via the `Feeds::AppleSubscription` instance will decouple the join from the apple specific filtering that happens in `Apple::Show`. Off the top of my head, things like archived episodes will not fall into the "not found" branch if we do that. (Assuming that`feed.episodes` and `feed.feed_episodes` would not be the same right? e.g. in the case of feed limits)

It seems like we'd want the Episode -> Apple::Episode join done in one place -- and avoid duplicating the filtering logic.

Reply:
ok, I am going to wait a bit on this for the other `Feeds::AppleSubscription` work to get merged, as it seems like that will simplify knowing if this episode ought to or will eventually be sent to Apple.